### PR TITLE
CSVS-93 Implementation of EventHandler was changed

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

File: src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandler.scala [hunk 1]
```diff
@@ -37,69 +40,111 @@ import scala.util.{Failure, Success, Try}
...
-        false
-      } else {
-        event.action match {
-          case Some(action) if action.oneOf(AccountCreate, AccountDelete, VMCreate, VMDelete) =>
-            event.status.getOrElse(Other) == CloudStackEvent.Status.Completed
-          case _ =>
-            false
-        }
-      }
+  private def runEventHandle(leaderLatch: InterruptableCountDawnLatch,
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandler.scala]: runEventHandling

File: src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandler.scala [hunk 2]
```diff
@@ -37,69 +40,111 @@ import scala.util.{Failure, Success, Try}
...
+                             event: CloudStackEvent,
+                             entityId: UUID,
+                             eventHandleFunc: (UUID) => Unit): Unit = {
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandler.scala]: eventHandlingFunc

File: src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala [hunk 3]
```diff
@@ -117,18 +119,101 @@ class CloudStackEventHandlerTestSuite extends FlatSpec with TestData with BaseTe
...
+          throw new JsonParseException(jsonFactory.createParser(value), "")
+        }
+      }
+    }
+
+    assert(cloudStackEventHandler.handle(dummyFlag).size == records.size)
+  }
+
+  "handle" should "change input flag to false and return empty OutputEnvelopes list " +
+    "if controller thrown an exception" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala]: fix description (English)

File: src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala [hunk 4]
```diff
@@ -117,18 +119,101 @@ class CloudStackEventHandlerTestSuite extends FlatSpec with TestData with BaseTe
...
+          throw new Exception
+        }
+      }
+    }
+
+    assert(cloudStackEventHandler.handle(dummyFlag).isEmpty)
+    assert(!dummyFlag.get())
+  }
+
+  "handle" should "handle event such as unknown event if JsonSerializer thrown JsonParseException" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala]: fix description (English)

File: src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala [hunk 5]
```diff
@@ -117,18 +119,101 @@ class CloudStackEventHandlerTestSuite extends FlatSpec with TestData with BaseTe
...
+      messageCount = 10,
+      controller
+    )
 
-    assert(cloudStackEventHandler.handleEventsFromRecords(records).isEmpty)
+    assert(cloudStackEventHandler.handle(dummyFlag).size == records.size)
   }
 
-  "restartEvent" should "handle event" in {
+  "handle" should "change input flag to false and return empty outputEnvelopes list " +
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala]: fix description (English)

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandler.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandler.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandler.scala (hunk 2 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandler.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala (hunk 5 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala (hunk 4 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala (hunk 3 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandlerTestSuite.scala]: ok

**Review by placccebo — APPROVED** [review]

**Pull request merged**